### PR TITLE
Additional engine flex documentation updates

### PR DESCRIPTION
**Pull request by lyaunzbe**
Additional engine flex documentation:

- added an item to the faqs doc
- added a new page with a deeper dive into the technical details of work with a engine flex project

### CODE REVIEW

**Review by Jbern16 — APPROVED** [review]

File: art-blocks-engine-onboarding/art-blocks-engine-101/Engine-technical-details.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+function updateProjectExternalAssetDependency(uint256 _projectId, uint256 _index, string calldata _cid, ExternalAssetDependencyType _dependencyType)
+function removeProjectExternalAssetDependency(uint256 _projectId, uint256 _index)
+```
+
+Some important notes to keep in mind with the above functions:
+- Only allowlisted/artist addresses can call these.
+- `ExternalAssetDependencyType _dependencyType` is a solidity enum, which can be passed in to these functions as a uint8. This enum only defines two options as of now, `IPFS` and `ARWEAVE`, which can be represented as `0` and `1` respectively. 
+- In the interest of saving gas, the `removeProjectExternalAssetDependency()` function is implemented in such a way that it does not preserve the order of the project's external asset dependency mapping and, therefore, the order in which the external asset depdendencies are returned should not be relied upon in any way. Additional implementation details described here: https://github.com/ArtBlocks/artblocks-contracts/blob/main/contracts/PBAB%2BCollabs/GenArt721CoreV2_ENGINE_FLEX.sol#L596
```
Comment (jakerockland) [inline art-blocks-engine-onboarding/art-blocks-engine-101/Engine-technical-details.md:33]: @lyaunzbe Can we chat more about this? Reading this has me realizing that I think we need to update our contract logic, as I'm not sure that the logic implemented as-is is really what we want here.

**Review by jakerockland — APPROVED** [review]
Generally all LGTM, this is awesome 🙏 🔥 

Added a commit with some minor nits and then left a note on one bigger thing I want to dig into together.

File: art-blocks-engine-onboarding/art-blocks-engine-101/Engine-technical-details.md (hunk 1 above)
Comment (lyaunzbe) [reply art-blocks-engine-onboarding/art-blocks-engine-101/Engine-technical-details.md:33]: This was discussed and decision was made to do the following:
- current logic external asset dep removal logic remains
- adding more documentation around this removal logic to ensure it's very clear
- followup: updating subgraph remove handler to check and remove empty elements in the mapping. this also sets on the path of easily forking the flex contract in the future with an updated removal function that leaves gaps in the mapping

Comment (jakerockland) [reply art-blocks-engine-onboarding/art-blocks-engine-101/Engine-technical-details.md:33]: +++ 🙏 💜 🙏 💜 🙏 💜 🙏 💜

**Comment by jakerockland** [comment]
Rather than waiting for explicit LGTM from @ABDruid and/or @cooper-temple , would say to ship-it and merge this once you feel good with the state of the docs @lyaunzbe and we can continue to iterate based on feedback from the Dan/Cooper on tweaking the content here. 💜

**Pull request merged**